packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +39,176 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
+  }
+  if (service === 'ElasticContainer') {
+    cfnTemplate.Resources.CustomExecutionPolicyForContainer = customExecutionPolicy;
+  }
+
+  await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+}
+
+
+//validate the format of actions and ARNs for custom IAM policies
+function validateCustomPolicy(customPolicy: CustomIAMPolicy, category: string, resourceName: string) {
+  const resources = customPolicy.Resource;
+  const actions = customPolicy.Action;
+  const resourceRegex = new RegExp('arn:(aws[a-zA-Z0-9-]*):([a-zA-Z0-9\\-])+:([a-z]{2}(-gov)?-[a-z]+-\\d{1})?:(\\d{12})?:(.*)');
+  const actionRegex = new RegExp('[a-zA-Z0-9]+:[a-z|A-Z|0-9|*]+');
+  const wrongResourcesRegex = [];
+  const wrongActionsRegex = [];
+  let errorMessage = "";
+
+  for (const resource of resources) {
+    if (!(resourceRegex.test(resource) || resource === '*')) {
+      wrongResourcesRegex.push(resource);
+    }
+  }
+
+  for (const action of actions) {
+    if (!actionRegex.test(action)) {
+      wrongActionsRegex.push(action);
+    }
+  }
+
+  const customPoliciesPath = pathManager.getCustomPoliciesPath(category, resourceName);
+
+  if (wrongResourcesRegex.length > 0) {
+    errorMessage += `Invalid custom IAM policy for {function name}. Incorrect "Resource": ${wrongResourcesRegex.toString()}\n Edit ${customPoliciesPath} to fix`;
+  }
+  if (wrongActionsRegex.length > 0) {
+    errorMessage += `Invalid custom IAM policy for {function name}. Incorrect "Action": ${wrongActionsRegex.toString()}\n Edit ${customPoliciesPath} to fix`;
+  }
+
+  if (errorMessage.length > 0) {
+    throw new CustomPoliciesFormatError(errorMessage);
+  }
+}
+
+function validateExistCustomPolicies(customPolicies: CustomIAMPolicies) : Boolean {
+  const ajv = new Ajv();
+  const validatePolicies = ajv.compile(CustomIAMPoliciesSchema);
+
+
+  if (!validatePolicies(customPolicies)) {
+    return false;
+  }
+
+  if (customPolicies.length === 1
+    && customPolicies[0].Action?.length === 0
+    && customPolicies[0].Resource?.length === 0) {
+      return false;
+    }
+
+  return true;
+}
+
+//replace or add env parameter in the front of the resource customers enter to the current env
+async function replaceEnvForCustomPolicies(policy: CustomIAMPolicy, currentEnv: string) : Promise<CustomIAMPolicy> {
+  const resourceWithEnv = [];
+  const action = policy.Action;
+  const effect = policy.Effect;
+  const customIAMpolicy: CustomIAMPolicy = {
+    Action: action,
+    Effect: effect,
+    Resource: []
+  }
+  for (let resource of policy.Resource){
+    if (resource.includes('${env}')) {
+      resourceWithEnv.push(Fn.Sub(resource, {'env': Fn.Ref('env')}));
+    }
+    else {
+      resourceWithEnv.push(resource);
+    }
+  }
+  customIAMpolicy.Resource = resourceWithEnv;
+  return customIAMpolicy;
+}
+
+
+async function validateCustomPoliciesSchema(data: CustomIAMPolicies, categoryName: string, resourceName: string) {
```

Comment:
this function can also go away once the json schema is updated

Reply:
This function is to validate if the customers enter the right schema, if not throw errors to them.